src/main/groovy/org/savantbuild/plugin/linter/LinterPlugin.groovy

```diff
@@ -54,9 +57,28 @@ class LinterPlugin extends BaseGroovyPlugin {
     boolean reportSuppressedViolations = attributes.getOrDefault("reportSuppressedViolations", false)
     String languageVersion = attributes.getOrDefault("languageVersion", "17")
     String inputPath = attributes.getOrDefault("inputPath", "src/main/java")
+    String sourceClassesPath = attributes.getOrDefault("sourceClassesPath", "build/classes/main")
+    String testClassesPath = attributes.getOrDefault("testClassesPath", "build/classes/test")
+    String cachePath = attributes.getOrDefault("cachePath", "build/pmd.cache")
     String minimumPriority = attributes.getOrDefault("minimumPriority", "MEDIUM")
     String[] ruleSets = attributes.getOrDefault("ruleSets", [])
     boolean failOnViolations = attributes.getOrDefault("failOnViolations", true)
+    String customRuleDependencyGroup = attributes.getOrDefault("customRuleDependencyGroup", null)
+    def ruleClassLoader = RuleSetLoader.class.getClassLoader()
+    if (customRuleDependencyGroup) {
+      def rules = new DependencyService.TraversalRules().with(customRuleDependencyGroup,
+          new DependencyService.TraversalRules.GroupTraversalRule(false, false))
+      def graph = project.dependencyService.resolve(project.artifactGraph, project.workflow, rules)
```

Comment:
The project using the linter plugin is what specifies the custom rule dependencies in a special dependency group and then passes that groupo here as a string.

One potential problem I see here is that Savant may then include that in the dependency graph for the running code (which is not actually what I want to happen). I'm not sure.

Reply:
Is there a way to be certain that this will not pollute the dependency graph in general, and only affect this plugin execution? 

Is the alternative to this to require all new PMD rules to be dded to this repo?

Reply:
I need to confirm whether Savant includes all groups in the graph or not. savant-core has some traversal rules that only specify certain groups.

If that doesn't bode well, or regardless, I could see about just supplying the dependency IDs directly and have the plugin resolve it then, without the `project` involved.

Reply:
@voidmain any other ideas on how to accomplish this, or any concerns?

Reply:
Savant loads all plugins in their own ClassLoaders. See https://github.com/savant-build/savant-core/blob/main/src/main/java/org/savantbuild/plugin/DefaultPluginLoader.java

That is where the Plugin implementation is instantiated from the plugin specific ClassLoader.

I don't know how the Linter works specifically, but if it just uses the source code, then it should be fine. If it needs some type of classpath or ClassLoader, that can get tricky because it might be in a different ClassLoader than the project is.

Reply:
@robotdan as I suspected, the way I had it could "pollute" the dep graph. I changed the code around to build a fresh graph just for these rules. Check out the latest update.

And also check out https://github.com/FusionAuth/fusionauth-app-linter-rules/commit/ffaaaf784569b9da19a23da5dae891a6b5f8461f#diff-238decee4eccd965c5c265b76431013b0225475a4eb1ba84e433a75c2bddf001R75 for how this is actually used